src/CsrfMiddleware.php

```diff
@@ -24,23 +23,19 @@ final class CsrfMiddleware implements MiddlewareInterface
     private string $headerName = self::HEADER_NAME;
 
     private ResponseFactoryInterface $responseFactory;
-    private CsrfTokenStorageInterface $storage;
+    private CsrfToken $token;
 
     public function __construct(
         ResponseFactoryInterface $responseFactory,
-        CsrfTokenStorageInterface $storage
+        CsrfToken $token
```

Comment:
```suggestion
        CsrfTokenService $tokenService
```

I think we should be clear that `CsrfToken` is **not** the data transfer object holding the token value.

Reply:
May be named `CsrfTokenManager` or something alike alternatively to "service".